### PR TITLE
*: Support group_concat

### PR DESCRIPTION
**Pull request by shenli**
@coocood @qiuyesuifeng

### CODE REVIEW

File: ast/functions.go [hunk 1]
```diff
@@ -423,6 +427,8 @@ func (n *AggregateFuncExpr) Update() error {
 		return n.updateMaxMin(false)
 	case AggFuncSum:
 		return n.updateSum()
+	case AggFuncGroupConcat:
```
Comment (qiuyesuifeng) [inline ast/functions.go]: Please keep in order.

Comment (shenli) [inline ast/functions.go]: There will be a pr after all the aggregate pr merged to do the clean up works. Or there will be conflicts.

File: ast/functions.go [hunk 2]
```diff
@@ -531,6 +537,39 @@ func (n *AggregateFuncExpr) updateSum() error {
...
+		}
+	}
+	var buf bytes.Buffer
+	if ctx.Value != nil {
+		// now use comma separator
+		buf.WriteString(ctx.Value.(string))
+		buf.WriteString(",")
+	}
+	for _, val := range vals {
+		buf.WriteString(fmt.Sprintf("%v", val))
```
Comment (qiuyesuifeng) [inline ast/functions.go]: Here is it OK to append all vals?

Comment (shenli) [inline ast/functions.go]: Yes, it is the same way with MySQL.
mysql> select * from tab0;
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    59
Current database: test

+------+------+------+
| col0 | col1 | col2 |
+------+------+------+
|    1 |    2 |    3 |
|    2 |    1 |    3 |
|    3 |    1 |    2 |
|    4 |    5 |    1 |
|    6 |    3 |    1 |
+------+------+------+
5 rows in set (0.02 sec)

mysql> select group_concat(col0, col1) from tab0 group by col2;
+--------------------------+
| group_concat(col0, col1) |
+--------------------------+
| 45,63                    |
| 31                       |
| 12,21                    |
+--------------------------+
3 rows in set (0.00 sec)

mysql>

**Comment by qiuyesuifeng** [comment]
@shenli  branch conflicted.

**Comment by shenli** [comment]
@qiuyesuifeng Solved

**Comment by qiuyesuifeng** [comment]
LGTM 
@coocood  PTAL

File: ast/functions.go [hunk 3]
```diff
@@ -532,6 +538,39 @@ func (n *AggregateFuncExpr) updateSum() error {
...
+	if n.Distinct {
+		d, err := ctx.distinctChecker.Check(vals)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if !d {
+			return nil
+		}
+	}
+	var buf bytes.Buffer
```
Comment (coocood) [inline ast/functions.go]: We can put buffer in context, so we don't need to create a new buffer on every update.

**Comment by shenli** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**